presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveCaching.java

```diff
@@ -32,6 +32,9 @@
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.testng.Assert.assertEquals;
 
+// make sure dedicated Rubix tests are executed first so that JMX counters are all set to 0
+// and no pending async read is in progress
+@Test(priority = -1)
```

Comment:
There might be IO operations caused by data preparations before tests start.
Instead of this, i would rather work on the test to be more resilient.

Maybe you can wait before the test start for any async IOs to complete? (there could be JMX counter with current async IOs)

or maybe you can change the assertions in the test?

Reply:
Made that test to execute exclusively.

> There might be IO operations caused by data preparations before tests start.

Inserts don't affect cache.

> Instead of this, i would rather work on the test to be more resilient.

Believe me. I tried. Rubix doesn't provide strict consistency for JMX counters in async read mode.

> Maybe you can wait before the test start for any async IOs to complete? (there could be JMX counter with current async IOs

There is such counter, but it's not reliable. You would still have to sleep.